src/Control/Exception/Annotated.hs

```diff
@@ -114,9 +117,6 @@ instance (Exception exception) => Exception (AnnotatedException exception) where
         | Just (e :: exception) <- Safe.fromException exn
         =
             pure $ new e
-        | Just x <- flatten <$> Safe.fromException exn
-        =
-            pure x
```

Comment:
I moved the flattening of exceptions into the `toException`, which works (per test suites)